distributed/worker.py

```diff
@@ -2037,9 +2022,9 @@ async def gather_dep(self, worker, dep, deps, total_nbytes, cause=None):
                 logger.exception("Worker stream died during communication: %s", worker)
                 self.log.append(("receive-dep-failed", worker))
                 for d in self.has_what.pop(worker):
-                    self.who_has[d].remove(worker)
-                    if not self.who_has[d]:
-                        del self.who_has[d]
+                    self.tasks[d].who_has.remove(worker)
+                # If the worker has died, `return` here so we don't hit `finally`
+                return
```

Comment:
??  What happened here?

Reply:
I genuinely don't remember why I thought this was necessary but tests pass locally when I remove it.